shopBuddy/shopBuddy/ItemDetailViewController.m

```diff
@@ -43,14 +55,26 @@ - (void)_callAPI {
 
 - (void)_populateView {
     //TODO: add Show More button and shortened description
-    self.titleLabel.text = self.item.name;
-    self.brandLabel.text = self.item.brand;
-    self.descriptionView.text = self.item.item_description;
+    titleLabel.text = self.item.name;
+    brandLabel.text = self.item.brand;
+    descriptionView.text = self.item.item_description;
     NSString *const URLString = self.item.images[0];
     NSURL *const url = [NSURL URLWithString:URLString];
-    [self.itemImage setImageWithURL:url];
+    [itemImage setImageWithURL:url];
 }
 
+- (void) _makeMenu {
+    NSMutableArray* actions = [[NSMutableArray alloc] init];
+    for (ShoppingList *list in lists) {
+        NSString *const actionTitle = [NSString stringWithFormat:@"Add Item to '%@' list", list.store_name];
+        [actions addObject:[UIAction actionWithTitle:actionTitle image:nil identifier:nil handler:^(__kindof UIAction* _Nonnull action) {
+            [ShoppingList addItemToList:self.item withList:list];
```

Comment:
You need to add a callback to this method so you only perform the navigation after the API call is complete to ensure the change is committed. You should add a spinner to the UI while you wait for the response.

Reply:
I'll add spinners to all the API calls (with progress bars for the longer ones) after the core features are implemented!